src/stcal/ramp_fitting/ramp_fit_class.py

```diff
@@ -0,0 +1,118 @@
+class RampFitInternal:
+    def __init__(self):
+        """
+        Creates an internal ramp fit class.
+        """
+        # Arrays from the data model
+        self.data = None
+        self.err = None
+        self.groupdq = None
+        self.pixeldq = None
+        self.int_times = None
+
+        # Meta information
+        self.instrume = None
+
+        self.frame_time = None
+        self.exp_ngroups = None
+        self.group_time = None
+        self.groupgap = None
+        self.nframes = None
+        self.drop_frames1 = None
+
+        # Data quality flags
+        self.flags_do_not_use = None
+        self.flags_jump_det = None
+        self.flags_saturated = None
+        self.flags_no_gain_val = None
+        self.flags_unreliable_slope = None
+
+    def set_arrays(self, data, err, groupdq, pixeldq, int_times):
```

Comment:
I have been told the pieces in `jwst` and `roman` will be the same and have the same variable names, so the a data model can be passed to `ramp_fit`.  If that is not the case, then the creation of this class to be used in ramp fitting will need to be pushed to the step code.

The `__init__` function defines all the necessary variables.  The class variables are set using methods for a couple of reasons.  There are 16 variables.  Having 16 input parameters into a single function is cumbersome and hard to read.  The class methods set variables of similar types.  This approach makes it more flexible to use in different pipelines, should this class instantiation need to be pushed back to step code, depending on how and where these pieces of information are defined in a particular pipeline.

The class can be pickled.  It needs to be able to be pickled in order to for mulitprocessing.  Since the class can be pickled disassembly and reassembly of the components of the class are unnecessary.

Reply:
As I understand it, not all of the jwst and roman pieces are the same. For example, instead of  JWST 'filter', Roman has 'optical_element'.

Reply:
I think the point is that this class represents the data needed in ramp fitting only, not the entire data model. Two different data models can be passed in but as long as they have `data, err, groupdq, pixeldq, int_times` .
That said, if there are differences in the (meta)data the Roman converters can take care of making sure they are the same, e.g. `optical_element` can be turned into `filter`.

Reply:
The code is designed such that a data model can be passed to `ramp_fit`, so long as it contains the proper values and variable names.  The creation and population of the class can occur in the step code with the step code calling `ramp_fit_internal`.  I changed the name from `ramp_fit_internal` to ramp_fit_data`, since that function may not only be used internally within the ramp fitting code.